apple-silicon-support/modules/peripheral-firmware/default.nix

```diff
@@ -43,20 +44,15 @@
 
     peripheralFirmwareDirectory = lib.mkOption {
       type = lib.types.nullOr lib.types.path;
-      default = let
-        paths = [
+
+      default = lib.findFirst (path: builtins.pathExists (path + "/all_firmware.tar.gz")) null
+        [
           # path when the system is operating normally
           "/boot/asahi"
           # path when the system is mounted in the installer
           "/mnt/boot/asahi"
         ];
 
-        validPaths = (builtins.filter
-          (p: builtins.pathExists (p + "/all_firmware.tar.gz"))
-          paths) ++ [ null ];
-
-        firstPath = builtins.elemAt validPaths 0;
-      in if firstPath != null then "${/. + firstPath}" else null;
```

Comment:
This has needs to be something like
```
default = let
  firmwarePath = lib.findFirst (path: builtins.pathExists (path + "/all_firmware.tar.gz")) null
        [
          # path when the system is operating normally
          "/boot/asahi"
          # path when the system is mounted in the installer
          "/mnt/boot/asahi"
        ];
  in if firmwarePath != null then "${/. + firmwarePath}" else null;
```
so that the firmware is copied into the store and so that we can cope with a missing firmware.

Reply:
I think just removing the quotes will have the intended effect:
```nix
      default = lib.findFirst (path: builtins.pathExists (path + "/all_firmware.tar.gz")) null
        [
          # path when the system is operating normally
          /boot/asahi
          # path when the system is mounted in the installer
          /mnt/boot/asahi
        ];
```
(since the purpose of `/. +` seems to just be to convert from a string to a path)

Reply:
This solution seems to work too. Thanks for the suggestion.